Modules/VMware.Skyline.InsightsApi/VMware.Skyline.InsightsApi.psm1

```diff
@@ -0,0 +1,397 @@
+#>
+    [cmdletbinding()]
+    param(
+        [Parameter(Mandatory=$true, ValueFromPipeline=$true, ValueFromPipelineByPropertyName=$true)][string]$findingId,
+        [Parameter(Mandatory=$true, ValueFromPipeline=$true, ValueFromPipelineByPropertyName=$true,ParameterSetName='ByObject')][System.Object]$products,
+        [Parameter(Mandatory=$true, ValueFromPipeline=$true, ValueFromPipelineByPropertyName=$true,ParameterSetName='ByName')][string]$product,
+        [Parameter(ParameterSetName='ByName')][string]$separator = '; ',
+        [ValidateRange(1,200)][int]$pagesize=200
+    )
+
+    begin {
+        $queryBody = @"
+        {
+            activeFindings(
+                filter: {
+                findingId: "$findingId",
+                product: "",
+                }) {
+            findings {
+                totalAffectedObjectsCount
+                affectedObjects(start: 0, limit: $pagesize)  {
+                    sourceName
+                    objectName
+                    objectType
+                    version
+                    buildNumber
+                    solutionTags {
+                    type
+                    version
+                    }
+                    firstObserved
+                }
+            }
+            totalRecords
+            timeTaken
+            }
+        }
+"@
+
+        # Try to get results the first time
+        $results = @()
+    }
+
+    process {
+        if ($PSCmdlet.ParameterSetName -eq 'ByName') {
+            $procProduct = $product.Split($separator).Trim() | Where-Object {$_ -ne '' }
+            # depending on separator we could end up with null rows on split, we'll omit those with where object
+            # to eliminate possible errors from the API
+        }
+
+        if ($PSCmdlet.ParameterSetName -eq 'ByObject') {
+            $procProduct = $products.products
+        }
+        
+        foreach ( $thisProduct in $procProduct ) {
+            $thisQueryBody = $queryBody -Replace 'findingId: "",', "findingId: `"$findingId`","
+            $thisQueryBody = $thisQueryBody -Replace 'product: "",', "product: `"$thisProduct`","
+            $thisIteration = 0
+
+            do {
+                $thisQueryBody = $thisQueryBody -Replace 'start: 0', "start: $thisIteration"
+                Write-Debug $thisQueryBody
+                $thisResult = Invoke-SkylineInsightsApi -queryBody (@{'query' = $thisQueryBody} | ConvertTo-Json -Compress)
+                $totalRecords = $thisResult.data.activeFindings.Findings.totalAffectedObjectsCount
+                $results += ($thisResult.data.activeFindings.Findings.affectedObjects) | Select-Object @{N='findingId';E={$findingId}}, *
+                $thisIteration += $pagesize
+            } while ($results.count -lt $totalRecords ) # end do/while loop
+        } # end foreach product loop
+    } 
+
+    end {
+        return $results
+    }
+}
+
+Function Format-SkylineResult {
```

Comment:
I don't think that this cmdlet is needed at all. Users might want to format differently, may only need certain properties etc., so this doesn't really bring any value. Date format that the API uses however is concerning. I would either add a helper function just for date conversion or would convert the output of the two cmdlets to fix the date.

Reply:
My thinking on this function was to help folks who wanted to export Skyline Findings to CSV.  By default, a direct `Get-SkylineFinding | Export-Csv filename.csv -NoTypeInformation` would result in many columns that had `System.Object[]` as the value.  The goal of this function was to try and return more friendly delimited strings that could easily be written to CSV.  Instead of using this separate format function, would it be more logical to have some sort of `-output` parameter on the `Get-*` functions with values like `raw, formatted` or similar?

Reply:
We normally leave it to the end user to format the output the way they need it. There are virtually limitless options on how customers may want their output to be formatted, so I think we should leave it entirely up to them. We expect our customers to have enough PowerShell knowledge to be able to format the output of a cmdlet.

Reply:
The Get-Skyline* functions would return the raw output that could be used in the limitless ways you describe.  However, for the normal/quick "I just want everything in a simple CSV" this optional function provides a sort of easy button.  I can remove the function if you have strong feelings one way or the other, but think it provides some value as is.

Reply:
Ok, if you feel it's valuable you can leave it.